resource.bnf

```diff
@@ -0,0 +1,31 @@
+;; basic types (u256, byte, length) are defined in common.bnf
+
+;; all fields without _cmtmnt (commitment) are non-lossy encodings of a pre-image into a finite field 
+;; 224 bytes
+resource ::= <logic-cmtmnt> <label> <quantity> <value-cmtmnt> <nonce> <nc> <ephemeral>
+
+logic_cmtmnt ::= <finite-field-element>
+
+label ::= <finite-field-element>
+
+quantity ::= <finite-field-element>
+
+value_cmtmnt ::= <finite-field-element>
+
+nonce ::= <finite-field-element>
+
+nc ::= <finite-field-element>
+
+ephemeral ::= <finite-field-element>
+
+;; 32 bytes
+finite-field-element ::= <u256>
+
+;; logic and value commitments are unique (up to hash collision) identifiers for program field elements
+logic ::= <program-field-element>
+
+value ::= <program-field-element>
+
+;; element of a variable size field
+;; 4 bytes + n bytes
+program-field-element ::= <length> <byte>*
```

Comment:
I don't see these used  in the resource data type at all, maybe belong elsewhere?

Reply:
These would be the plaintext representations of the field elements, which are the things the "VM" would need to load into memory if it were to operate on the resources, e.g. computing proofs and such.

Another way to see it is: The finite-field-elements are used as keys in a lookup table to retrieve the program-field-elements. 

If you think we should move that to another file, we can do that, to me it just seemed most usefully grouped with the resource used to communicate their keys.

Reply:
Might be good to write it in a readme or give a sister file, that sort of explains why better.

Thanks for writing up the BNF

Reply:
We can move these to data-storage.bnf